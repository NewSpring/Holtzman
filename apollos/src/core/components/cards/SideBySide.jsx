import { Component, PropTypes} from "react"
import { Link } from "react-router"
import { css } from "aphrodite";

import { ImageLoader } from "../loading"

import Styles from "../loading/FeedItemSkeleton-css"

let Wrapper = (props) => (
  <div {...this.props}>
    {this.props.children}
  </div>
)

export default class Card extends Component {

  static propTypes = {
    classes: PropTypes.array,
    theme: PropTypes.string,
    link: PropTypes.string,
    fallbackImage: PropTypes.string,
    images: PropTypes.array,
    styles: PropTypes.object,
  }

  itemClasses = () => {
    let classes = [
      "card__item",
      "soft",
      "text-left",
      "soft-double@anchored",
      "one-whole@palm",
      "one-whole@lap",
      "three-fifths@lap-wide",
      "three-fifths@palm-wide",
      "one-half@anchored",
    ];

    if (this.props.itemClasses) {
      classes = classes.concat(this.props.itemClasses);
    }

    return classes.join(" ");
  }

  cardClasses = () => {
    let classes = [
      "card",
    ];

    if (this.props.classes) {
      classes = classes.concat(this.props.classes);
    }

    return classes.join(" ");
  }

  styles = () => {
    let defaultStyles = {
      overflow: "hidden"
    }

    // if (this.props.image && this.props.image.full) {
    //   defaultStyles.backgroundImage = `url('${this.props.image.url}')`
    // }

    return defaultStyles
  }

  // context from ImageLoader
  preloader() {
    return (
      <div className={`${this.imageclasses.join(" ")} ${css(Styles["load-item"])}`}>
        <div className="ratio__item"></div>
      </div>
    );
  }

  // context from ImageLoader
  renderElement() {
    return (
      <div className={this.imageclasses.join(" ")} style={this.style}>
        <div className="ratio__item"></div>
      </div>
    );
  }

  getResponsiveImage = () => {
    const { images } = this.props;

    if (typeof window === "undefined" || window === null || !images) {
      return false;
    }

    let sizes = {};
    for (let image of images) {
      image.url = image.cloudfront ? image.cloudfront : image.s3;
      sizes[image.fileLabel] = image;
    }

    if (window.matchMedia("(max-width: 480px)").matches) {
      src = sizes["2:1"] ? sizes["2:1"].url : false
    } else if (window.matchMedia("(max-width: 730px)").matches) {
      src = sizes["1:2"] ? sizes["1:2"].url : false;
    } else if (window.matchMedia("(max-width: 768px)").matches) {
      src = sizes["1:1"] ? sizes["1:1"].url : false;
    } else if (window.matchMedia("(max-width: 1024px)").matches) {
      src = sizes["2:1"] ? sizes["2:1"].url : false;
    } else if (window.matchMedia("(max-width: 1281px)").matches) {
      src = sizes["1:2"] ? sizes["1:2"].url : false;
    } else {
      src = sizes["1:1"] ? sizes["1:1"].url : false;
    }

    return src;
  }

  createImage = () => {
    let { defaultImage, ratio } = this.props

    let imageclasses = [
      "background--fill",
      "card__image",
      "locked-ends@lap-wide-and-up",
      "locked-sides@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-sides@palm-wide",
      "relative@palm",
      "relative@lap",
    ];

    if (!ratio) ratio = "ratio--landscape"
    imageclasses.push(ratio);

    let src = this.getResponsiveImage() || defaultImage;

    let style = { backgroundImage: `url('${src}')` }

    return (
      <ImageLoader
        src={src}
        preloader={this.preloader}
        renderElement={this.renderElement}
        imageclasses={imageclasses}
        style={style}
      />
    )
  }


  render () {
    const { link, theme, styles, itemTheme, itemStyles } = this.props


    let wrapperClasses = [
      "relative@lap",
      "relative@palm",
      "plain",
      "locked-ends@lap-wide-and-up",
      "locked-right@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-right@palm-wide",
      "one-whole@lap",
      "one-whole@palm",
      "two-fifths@lap-wide",
      "two-fifths@palm-wide",
      "one-half@anchored"
    ].join(" ")


    if (link) {
      return (
        <div
          className={theme || this.cardClasses()}
          style={styles || this.styles() }
          >
          <Link className={wrapperClasses} to={link}>
            {this.createImage()}
          </Link>
          <div
            className={ itemTheme || this.itemClasses() }
            style={itemStyles}
          >
            {this.props.children}
          </div>

        </div>
      )
    }

    return (
      <div
        className={theme || this.cardClasses()}
        style={styles || this.styles() }
        >
        <div className={wrapperClasses}>
          {this.createImage()}
        </div>
        <div
          className={ itemTheme || this.itemClasses() }
          style={itemStyles}
        >
          {this.props.children}
        </div>

      </div>

    );

  }

}
