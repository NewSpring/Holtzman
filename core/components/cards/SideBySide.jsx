import { Component, PropTypes} from "react"
import { Link } from "react-router"

import { ImageLoader } from "../loading"

import Styles from "../loading/FeedItemSkeleton.css"

let Wrapper = (props) => (
  <div {...this.props}>
    {this.props.children}
  </div>
)

export default class Card extends Component {

  static propTypes = {
    classes: PropTypes.array,
    theme: PropTypes.string,
    link: PropTypes.string,
    image: PropTypes.object,
    styles: PropTypes.object
  }

  itemClasses = () => {
    let classes = [
      "card__item",
      "soft",
      "text-left",
      "soft-double@anchored",
      "one-whole",
      "three-fifths@lap-wide",
      "one-half@anchored",
    ];

    if (this.props.itemClasses) {
      classes = classes.concat(this.props.itemClasses);
    }

    return classes.join(" ");
  }

  cardClasses = () => {
    let classes = [
      "card",
    ];

    if (this.props.classes) {
      classes = classes.concat(this.props.classes);
    }

    return classes.join(" ");
  }

  styles = () => {
    let defaultStyles = {
      overflow: "hidden"
    }

    if (this.props.image && this.props.image.full) {
      defaultStyles.backgroundImage = `url(${this.props.image.url})`
    }

    return defaultStyles
  }

  // context from ImageLoader
  preloader() {
    return (
      <div className={`${this.imageclasses.join(" ")} ${Styles["load-item"]}`}>
        <div className="ratio__item"></div>
      </div>
    );
  }

  // context from ImageLoader
  renderElement() {
    return (
      <div className={this.imageclasses.join(" ")} style={this.style}>
        <div className="ratio__item"></div>
      </div>
    );
  }

  createImage = () => {

      const { image } = this.props

      if (image) {
        let imageclasses = [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "relative@lap"
        ]

        if (image.ratio) {
          imageclasses.push(`ratio--${image.ratio}`)
        } else {
          imageclasses.push("ratio--landscape")
        }


        let src = image.defaultImage
          if (typeof window != "undefined" && window != null) {
             if (window.matchMedia("(max-width: 768px)").matches) {
               src = image["2:1"] ? image["2:1"] : image.url
             } else if (window.matchMedia("(max-width: 1281px)").matches) {
               src = image["1:2"] ? image["1:2"] : image.url
             } else {
               src = image["1:1"] ? image["1:1"] : image.url
           }

         let style
         if (image.full != true) {
           style = {
             backgroundImage: `url(${src})`,
           }
         }

         console.log(image)
        return (
          <ImageLoader
            src={src}
            preloader={this.preloader}
            renderElement={this.renderElement}
            imageclasses={imageclasses}
            style={style}
          />
        )
      }
    }
  }


  render () {

    const { link, image, theme, styles, itemTheme, itemStyles } = this.props


    let wrapperClasses = "relative@lap plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up one-whole@lap two-fifths@lap-wide one-half@anchored"

    if (link) {
      return (
        <div
          className={theme || this.cardClasses()}
          style={styles || this.styles() }
          >
          <Link className={wrapperClasses} to={link}>
            {this.createImage()}
          </Link>
          <div
            className={ itemTheme || this.itemClasses() }
            style={itemStyles}
          >
            {this.props.children}
          </div>

        </div>
      )
    }

    return (
      <div
        className={theme || this.cardClasses()}
        style={styles || this.styles() }
        >
        <div className={wrapperClasses}>
          {this.createImage()}
        </div>
        <div
          className={ itemTheme || this.itemClasses() }
          style={itemStyles}
        >
          {this.props.children}
        </div>

      </div>

    )
  }

}
