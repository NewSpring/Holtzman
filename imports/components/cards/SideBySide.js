import { Component, PropTypes } from "react";
import { Link } from "react-router";
import { css } from "aphrodite";

import { ImageLoader } from "../loading";

import Styles from "../loading/FeedItemSkeleton-css";

export default class Card extends Component {

  static propTypes = {
    classes: PropTypes.array, // eslint-disable-line
    theme: PropTypes.string,
    link: PropTypes.string,
    itemClasses: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.array,
    ]),
    images: PropTypes.array, // eslint-disable-line
    styles: PropTypes.object, // eslint-disable-line
    ratio: PropTypes.string,
    defaultImage: PropTypes.string,
    itemTheme: PropTypes.string,
    itemStyles: PropTypes.object, // eslint-disable-line
    children: PropTypes.any, //eslint-disable-line
  }

  itemClasses = () => {
    let classes = [
      "card__item",
      "soft",
      "text-left",
      "soft-double@anchored",
      "one-whole@palm",
      "one-whole@lap",
      "three-fifths@lap-wide",
      "three-fifths@palm-wide",
      "one-half@anchored",
    ];

    if (this.props.itemClasses) {
      classes = classes.concat(this.props.itemClasses);
    }

    return classes.join(" ");
  }

  cardClasses = () => {
    let classes = [
      "card",
    ];

    if (this.props.classes) {
      classes = classes.concat(this.props.classes);
    }

    return classes.join(" ");
  }

  styles = () => {
    const defaultStyles = {
      overflow: "hidden",
    };

    // if (this.props.image && this.props.image.full) {
    //   defaultStyles.backgroundImage = `url('${this.props.image.url}')`
    // }

    return defaultStyles;
  }

  // context from ImageLoader
  preloader() {
    return (
      <div className={`${this.imageclasses.join(" ")} ${css(Styles["load-item"])}`}>
        <div className="ratio__item" />
      </div>
    );
  }

  // context from ImageLoader
  renderElement() {
    return (
      <div className={this.imageclasses.join(" ")} style={this.style}>
        <div className="ratio__item" />
      </div>
    );
  }

  getResponsiveImage = () => {
    const { images } = this.props;

    if (typeof window === "undefined" || window === null || !images) {
      return false;
    }

    const sizes = {};
    let src;
    for (const image of images) {
      sizes[image.fileLabel] = image;
    }

    if (window.matchMedia("(max-width: 480px)").matches) {
      src = sizes["2:1"] ? sizes["2:1"].url || sizes["2:1"].cloudfront : false;
    } else if (window.matchMedia("(max-width: 730px)").matches) {
      src = sizes["1:2"] ? sizes["1:2"].url || sizes["1:2"].cloudfront : false;
    } else if (window.matchMedia("(max-width: 768px)").matches) {
      src = sizes["1:1"] ? sizes["1:1"].url || sizes["1:1"].cloudfront : false;
    } else if (window.matchMedia("(max-width: 1024px)").matches) {
      src = sizes["2:1"] ? sizes["2:1"].url || sizes["2:1"].cloudfront : false;
    } else if (window.matchMedia("(max-width: 1281px)").matches) {
      src = sizes["1:2"] ? sizes["1:2"].url || sizes["1:2"].cloudfront : false;
    } else {
      src = sizes["1:1"] ? sizes["1:1"].url || sizes["1:1"].cloudfront : false;
    }

    return src;
  }

  createImage = () => {
    let { ratio } = this.props;
    const { defaultImage } = this.props;

    const imageclasses = [
      "background--fill",
      "card__image",
      "locked-ends@lap-wide-and-up",
      "locked-sides@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-sides@palm-wide",
      "relative@palm",
      "relative@lap",
    ];

    if (!ratio) ratio = "ratio--landscape";
    imageclasses.push(ratio);

    const src = this.getResponsiveImage() || defaultImage;

    const style = { backgroundImage: `url('${src}')` };

    return (
      <ImageLoader
        src={src}
        preloader={this.preloader}
        renderElement={this.renderElement}
        imageclasses={imageclasses}
        style={style}
      />
    );
  }


  render() {
    const { link, theme, styles, itemTheme, itemStyles } = this.props;


    const wrapperClasses = [
      "relative@lap",
      "relative@palm",
      "plain",
      "locked-ends@lap-wide-and-up",
      "locked-right@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-right@palm-wide",
      "one-whole@lap",
      "one-whole@palm",
      "two-fifths@lap-wide",
      "two-fifths@palm-wide",
      "one-half@anchored",
    ].join(" ");


    if (link) {
      return (
        <div
          className={theme || this.cardClasses()}
          style={styles || this.styles()}
        >
          <Link className={wrapperClasses} to={link}>
            {this.createImage()}
          </Link>
          <div
            className={itemTheme || this.itemClasses()}
            style={itemStyles}
          >
            {this.props.children}
          </div>

        </div>
      );
    }

    return (
      <div
        className={theme || this.cardClasses()}
        style={styles || this.styles()}
      >
        <div className={wrapperClasses}>
          {this.createImage()}
        </div>
        <div
          className={itemTheme || this.itemClasses()}
          style={itemStyles}
        >
          {this.props.children}
        </div>

      </div>

    );
  }

}
