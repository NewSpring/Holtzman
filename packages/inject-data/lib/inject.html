<script type="text/inject-data"><%= data %></script>
