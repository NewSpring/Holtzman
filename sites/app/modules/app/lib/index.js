

// named collections in EE :(
import Collections from "./collections"

export default {
  Collections
}
