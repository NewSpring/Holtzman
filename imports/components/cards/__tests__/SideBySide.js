import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { reset, startBuffering } from "aphrodite/lib/inject";

import SideBySide from "../SideBySide";

const defaultProps = {
  classes: [],
  theme: null,
  link: null,
  itemClasses: [],
  images: [
    { fileLabel: "2:1", url: "http://test.com/2x1.jpg" },
    { fileLabel: "1:1", url: "http://test.com/1x1.jpg" },
    { fileLabel: "1:2", url: "http://test.com/1x2.jpg" },
  ],
  styles: null,
  ratio: null,
  defaultImage: "http://test.com/default.jpg",
  itemTheme: null,
  itemStyles: {
    color: "red",
  },
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return (
    <SideBySide { ...newProps }>
      <h1>test</h1>
    </SideBySide>
  );
};

beforeEach(() => {
  reset();
  startBuffering();
});

afterEach(() => {
  reset();
});

it("should render with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("overrides with theme", () => {
  const wrapper = shallow(generateComponent({
    theme: "override me",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("overrides with styles", () => {
  const wrapper = shallow(generateComponent({
    styles: {
      color: "orange",
    },
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("overrides item theme", () => {
  const wrapper = shallow(generateComponent({
    itemTheme: "myItem",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders link version", () => {
  const wrapper = shallow(generateComponent({
    link: "http://test.com/link",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders link version with theme override", () => {
  const wrapper = shallow(generateComponent({
    link: "http://test.com/link",
    theme: "override link",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders link version with style override", () => {
  const wrapper = shallow(generateComponent({
    link: "http://test.com/link",
    styles: {
      color: "blue",
    },
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders link version with item theme override", () => {
  const wrapper = shallow(generateComponent({
    link: "http://test.com/link",
    itemTheme: "myOtherItem",
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("itemClasses renders defaults", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().itemClasses()).toMatchSnapshot();
});

it("itemClasses appends additional item classes", () => {
  const wrapper = shallow(generateComponent({
    itemClasses: ["one", "two"],
  }));
  expect(wrapper.instance().itemClasses()).toMatchSnapshot();
});

it("cardClasses renders defaults", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().cardClasses()).toMatchSnapshot();
});

it("cardClasses appends additional item classes", () => {
  const wrapper = shallow(generateComponent({
    classes: ["three", "four"],
  }));
  expect(wrapper.instance().cardClasses()).toMatchSnapshot();
});

it("styles returns default", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().styles()).toEqual({
    overflow: "hidden",
  });
});

it("getResponsiveImage returns false if no images", () => {
  const wrapper = shallow(generateComponent({
    images: null,
  }));
  expect(wrapper.instance().getResponsiveImage()).toBe(false);
});

it("getResponsiveImage returns for all browser breakpoints", () => {
  const imageTypes = [
    ["(max-width: 480px)", "2:1"],
    ["(max-width: 730px)", "1:2"],
    ["(max-width: 768px)", "1:1"],
    ["(max-width: 1024px)", "2:1"],
    ["(max-width: 1281px)", "1:2"],
    // if no match found
    [false, "1:1"],
  ];
  imageTypes.map((image) => {
    window.matchMedia = jest.fn((media) => (
      { matches: media === image[0] }
    ));
    const wrapper = shallow(generateComponent());
    const result = wrapper.instance().getResponsiveImage();
    expect(result).toBe(defaultProps.images.filter((i) => i.fileLabel === image[1])[0].url);
  });
});

it("getResponsiveImage works with legacy cloudfront attribute", () => {
  const imageTypes = [
    ["(max-width: 480px)", "2:1"],
    ["(max-width: 730px)", "1:2"],
    ["(max-width: 768px)", "1:1"],
    ["(max-width: 1024px)", "2:1"],
    ["(max-width: 1281px)", "1:2"],
    // if no match found
    [false, "1:1"],
  ];
  const props = {
    images: [
      { fileLabel: "2:1", cloudfront: "http://test.com/2x1.jpg" },
      { fileLabel: "1:1", cloudfront: "http://test.com/1x1.jpg" },
      { fileLabel: "1:2", cloudfront: "http://test.com/1x2.jpg" },
    ],
  };
  imageTypes.map((image) => {
    window.matchMedia = jest.fn((media) => (
      { matches: media === image[0] }
    ));
    const wrapper = shallow(generateComponent(props));
    const result = wrapper.instance().getResponsiveImage();
    expect(result).toBe(props.images.filter((i) => i.fileLabel === image[1])[0].cloudfront);
  });
});


it("createImage renders default", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().createImage()).toMatchSnapshot();
});

it("createImage uses custom ratio class", () => {
  const wrapper = shallow(generateComponent({
    ratio: "myRatio",
  }));
  expect(wrapper.instance().createImage()).toMatchSnapshot();
});

it("createImage uses default image if no images", () => {
  const wrapper = shallow(generateComponent({
    images: null,
  }));
  expect(wrapper.instance().createImage()).toMatchSnapshot();
});
