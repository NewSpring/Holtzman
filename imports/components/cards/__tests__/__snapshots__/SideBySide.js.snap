exports[`test cardClasses appends additional item classes 1`] = `"card three four"`;

exports[`test cardClasses renders defaults 1`] = `"card"`;

exports[`test createImage renders default 1`] = `
<ImageLoader
  imageclasses={
    Array [
      "background--fill",
      "card__image",
      "locked-ends@lap-wide-and-up",
      "locked-sides@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-sides@palm-wide",
      "relative@palm",
      "relative@lap",
      "ratio--landscape",
    ]
  }
  preloader={[Function]}
  renderElement={[Function]}
  src="http://test.com/1x1.jpg"
  style={
    Object {
      "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
    }
  }
  wrapper={[Function]} />
`;

exports[`test createImage uses custom ratio class 1`] = `
<ImageLoader
  imageclasses={
    Array [
      "background--fill",
      "card__image",
      "locked-ends@lap-wide-and-up",
      "locked-sides@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-sides@palm-wide",
      "relative@palm",
      "relative@lap",
      "myRatio",
    ]
  }
  preloader={[Function]}
  renderElement={[Function]}
  src="http://test.com/1x1.jpg"
  style={
    Object {
      "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
    }
  }
  wrapper={[Function]} />
`;

exports[`test createImage uses default image if no images 1`] = `
<ImageLoader
  imageclasses={
    Array [
      "background--fill",
      "card__image",
      "locked-ends@lap-wide-and-up",
      "locked-sides@lap-wide-and-up",
      "locked-ends@palm-wide",
      "locked-sides@palm-wide",
      "relative@palm",
      "relative@lap",
      "ratio--landscape",
    ]
  }
  preloader={[Function]}
  renderElement={[Function]}
  src="http://test.com/default.jpg"
  style={
    Object {
      "backgroundImage": "url(\'http://test.com/default.jpg\')",
    }
  }
  wrapper={[Function]} />
`;

exports[`test itemClasses appends additional item classes 1`] = `"card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored one two"`;

exports[`test itemClasses renders defaults 1`] = `"card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"`;

exports[`test overrides item theme 1`] = `
<div
  className="card"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <div
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </div>
  <div
    className="myItem"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test overrides with styles 1`] = `
<div
  className="card"
  style={
    Object {
      "color": "orange",
    }
  }>
  <div
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </div>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test overrides with theme 1`] = `
<div
  className="override me"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <div
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </div>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test renders link version 1`] = `
<div
  className="card"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <Link
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="http://test.com/link">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </Link>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test renders link version with item theme override 1`] = `
<div
  className="card"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <Link
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="http://test.com/link">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </Link>
  <div
    className="myOtherItem"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test renders link version with style override 1`] = `
<div
  className="card"
  style={
    Object {
      "color": "blue",
    }
  }>
  <Link
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="http://test.com/link">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </Link>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test renders link version with theme override 1`] = `
<div
  className="override link"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <Link
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="http://test.com/link">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </Link>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`test should render with props 1`] = `
<div
  className="card"
  style={
    Object {
      "overflow": "hidden",
    }
  }>
  <div
    className="relative@lap relative@palm plain locked-ends@lap-wide-and-up locked-right@lap-wide-and-up locked-ends@palm-wide locked-right@palm-wide one-whole@lap one-whole@palm two-fifths@lap-wide two-fifths@palm-wide one-half@anchored">
    <ImageLoader
      imageclasses={
        Array [
          "background--fill",
          "card__image",
          "locked-ends@lap-wide-and-up",
          "locked-sides@lap-wide-and-up",
          "locked-ends@palm-wide",
          "locked-sides@palm-wide",
          "relative@palm",
          "relative@lap",
          "ratio--landscape",
        ]
      }
      preloader={[Function]}
      renderElement={[Function]}
      src="http://test.com/1x1.jpg"
      style={
        Object {
          "backgroundImage": "url(\'http://test.com/1x1.jpg\')",
        }
      }
      wrapper={[Function]} />
  </div>
  <div
    className="card__item soft text-left soft-double@anchored one-whole@palm one-whole@lap three-fifths@lap-wide three-fifths@palm-wide one-half@anchored"
    style={
      Object {
        "color": "red",
      }
    }>
    <h1>
      test
    </h1>
  </div>
</div>
`;
